pkg/controller/certificates/issuing/secret_manager.go

```diff
@@ -54,171 +53,43 @@ func (c *controller) ensureSecretData(ctx context.Context, log logr.Logger, crt
 		return err
 	}
 
-	// Check whether the Certificate's Secret has correct output format and
-	// metadata.
-	reason, message, isViolation := c.postIssuancePolicyChain.Evaluate(policies.Input{
-		Certificate: crt,
-		Secret:      secret,
-	})
+	log = log.WithValues("secret", secret.Name)
 
-	var data secretsmanager.SecretData
+	var data internal.SecretData
 	if secret.Data != nil {
-		data = secretsmanager.SecretData{
+		data = internal.SecretData{
 			PrivateKey:  secret.Data[corev1.TLSPrivateKeyKey],
 			Certificate: secret.Data[corev1.TLSCertKey],
 			CA:          secret.Data[cmmeta.TLSCAKey],
 		}
 	}
 
-	log = log.WithValues("secret", secret.Name)
-
-	// Check whether the Certificate's SecretTemplate matches that on the Secret.
-	secretTemplateMatchManagedFields, err := c.secretTemplateMatchesManagedFields(crt, secret, data)
-	if err != nil {
-		// An error here indicates that the managed fields are malformed and the
-		// decoder doesn't understand the managed fields on the Secret, or the
-		// signed certificate data could not be decoded. There is nothing more the
-		// controller can do here, so we exit nil so this controller doesn't end in
-		// an infinite loop.
-		log.Error(err, "failed to determine whether the SecretTemplate matches Secret")
-		return nil
-	}
+	// Check whether the Certificate's Secret has correct output format and
+	// metadata.
+	reason, message, isViolation := c.postIssuancePolicyChain.Evaluate(policies.Input{
+		Certificate: crt,
+		Secret:      secret,
+	})
 
-	// - secretTemplateMatchesSecret: If a key or value changed on the
-	// Annotations or Labels in the SecretTemplate, the SecretTemplate will not
-	// match the Annotations or Labels on the Secret.
-	// - secretTemplateMatchManagedFields: If a key was removed on the
-	// SecretTemplate, then the managed fields on the Secret won't match.
-	// In either case, the Secret needs to be re-reconciled with the Secrets
-	// Manager.
-	if !secretTemplateMatchesSecret(crt, secret) || !secretTemplateMatchManagedFields {
-		log.Info("mismatch between SecretTemplate and Secret, updating Secret annotations/labels")
-		return c.secretsUpdateData(ctx, crt, data)
+	if isViolation {
+		switch reason {
+		case policies.InvalidCertificate, policies.ManagedFieldsParseError:
```

Comment:
Note: I guess `Evaluate` could instead just return an error in these cases (since also I guess otherwise in future we might end up adding more 'error' reasons)

Reply:
Indeed, but I didn't want to change the Evaluate function signature at all in the refactor